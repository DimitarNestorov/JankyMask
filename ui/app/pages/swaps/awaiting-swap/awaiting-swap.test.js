import React from 'react';
import configureMockStore from 'redux-mock-store';

import {
  renderWithProvider,
  createSwapsMockStore,
} from '../../../../../test/jest';
import AwaitingSwap from '.';

const createProps = (customProps = {}) => {
  return {
    swapComplete: false,
    txHash: 'txHash',
    tokensReceived: 'tokensReceived',
    submittingSwap: true,
    inputValue: 5,
    maxSlippage: 3,
    ...customProps,
  };
};

describe('AwaitingSwap', () => {
  it('renders the component with initial props', () => {
    const store = configureMockStore()(createSwapsMockStore());
    const { getByText } = renderWithProvider(
      <AwaitingSwap {...createProps()} />,
      store,
    );
    expect(getByText('Processing')).toBeInTheDocument();
    expect(getByText('View on Etherscan')).toBeInTheDocument();
    expect(getByText('View in activity')).toBeInTheDocument();
    expect(
      document.querySelector('.awaiting-swap__main-descrption'),
    ).toMatchSnapshot();
    expect(document.querySelector('.swaps-footer')).toMatchSnapshot();
  });
});
