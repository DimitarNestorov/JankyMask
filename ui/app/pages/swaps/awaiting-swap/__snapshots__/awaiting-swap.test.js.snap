// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AwaitingSwap renders the component with initial props 1`] = `
<div
  class="awaiting-swap__main-descrption"
>
  <span>
     
    Your 
    <span
      class="awaiting-swap__amount-and-symbol"
    >
      ETH
    </span>
     will be added to your account once this transaction has processed.
     
  </span>
</div>
`;

exports[`AwaitingSwap renders the component with initial props 2`] = `
<div
  class="swaps-footer"
>
  <div
    class="swaps-footer__buttons"
  >
    <div
      class="page-container__footer swaps-footer__custom-page-container-footer-class"
    >
      <footer>
        <button
          class="button btn-primary page-container__footer-button swaps-footer__custom-page-container-footer-button-class swaps-footer__custom-page-container-footer-button-class--single"
          data-testid="page-container-footer-next"
          disabled=""
          role="button"
          tabindex="0"
        >
          View in activity
        </button>
      </footer>
    </div>
  </div>
</div>
`;
